### ISSUE TITLE
Trailing and leading quote characters are removed

### ISSUE THREAD

**Issue by mefcorvi** [yargs/yargs-parser#201]
Version 13.1.1

### Description

Trailing and leading quote characters are removed from the values. If I want to pass a `'string'` as a value I need to triple-quote it as `"''string''"` or `\'''string\'''` when I'm specifying that in the shell.

### Expected

```javascript
> require('yargs-parser')([ '--arg', '"option"' ])
{ _: [], arg: '"option"' }
```

### Actual

```javascript
> require('yargs-parser')([ '--arg', '"option"' ])
{ _: [], arg: 'option' }
```

This probably relates to #138, #180. Definitely relates to #145 and #153.
I would suggest introducing a new `keepQuotes`/`trimQuotes` option to support the all edge cases.

**Comment by bcoe** [comment]
@mefcorvi in an ideal world `my-app --argument \"foo\"` would work, might be worth rethinking this bit of the parser.

**Comment by shadowspawn** [comment]
I can no longer reproduce the reported issue. I suspect it was fixed by #407.
